### ISSUE TITLE
EJS missing from the list of languages, and the highlighting is incorrect.

### ISSUE THREAD

**Issue by thecatishungry** [Digitalbrainstem/ejs-grammar#40]
EJS language support: 1.1.3
VSCode: Version: 1.37.1 (system setup)
Commit: f06011ac164ae4dc8e753a3fe7f9549844d15e35
Date: 2019-08-15T16:17:55.855Z
Electron: 4.2.7
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 6.1.7601

First of all, sorry if I am missing something obvious here, but with the latest version I got a few problems.

EJS disappeared from the language select menu. I checked the author's twitter and there it says - "Major Changes - EJS Language Support is now based on HTML! EJS files now act as HTML files with EJS tags built in! All features of HTML files are supported." so I guess that is normal behavior?

The second problem is that the syntax highlighting seems really incorrect.

```
<div>
    <div class="fixtures">
        <% data.forEach(item => { %>
        <div class="fixture">
            <div><%- item.home_team_name %></div>
            <img src="<%- item.home_team_logo %>" alt="<%- item.home_team_name %>"></img>
            <img src="<%- item.away_team_logo %>" alt="<%- item.away_team_name %>"></img>
            <div><%- item.away_team_name %></div>
        </div>
        <% }); %>
    </div>
</div>
```

This produces the following: [https://i.gyazo.com/a40732b281410266a7a70456cac1ae81.png](https://i.gyazo.com/a40732b281410266a7a70456cac1ae81.png)

Since I only started learning EJS about a few days ago, I thought there might be a problem with my code, so I took some code right out of EJS's documentation and it produces the same problem:

```
<ul>
  <% users.forEach(function(user){ %>
    <%- include('user/show', {user: user}); %>
  <% }); %>
</ul>
```

Results in - [https://i.gyazo.com/deacfefeaf3d6d0c41d781c873dc3a43.png](https://i.gyazo.com/deacfefeaf3d6d0c41d781c873dc3a43.png)

I have also tried modifying my `settings.json` file, but no result.

```
{
    "files.associations": {
        "*.ejs": "html"
    }
}
```

Would really like if someone can point me out in the right direction of what I'm doing wrong here or if this is indeed a bug.

Thanks.

**Comment by thecatishungry** [comment]
> There is a bug with the functions that have arrow functions.

Are you sure it's related to arrow functions? For example, check this one:

```
<div>
    <% data.forEach(function(item) { %>
        <div><%- item %></div>
    <% } %>
</div>
```

[https://i.gyazo.com/a4e0646ab844029260a489f7df046434.png](https://i.gyazo.com/a4e0646ab844029260a489f7df046434.png)

A normal for-loop such as `for (let i=0; i<10; i++)` works but not a `forEach`.

I have also noticed another thing which might be worth checking out:

Same behavior as the example above where the closing `</div>` has the wrong color.

```
<div>
    <% let foo = 'bar' %>
    <span><%- foo %></span>
</div>
```

This doesn't work, however if we if there is no `let` or `var` it works as intended:

```
<div>
    <% foo = 'bar' %>
    <span><%- foo %></span>
</div>
```

Thanks a lot for your reply and the suggested solution. It's not of critical importance for me so I will stay with the current version and wait just for the updates whenever that may be.

Cheers.

**Comment by Betanu701** [comment]
@thecatishungry The orginal errors have been fixed as of version 1.2. I am going to close this.

As for the other bug you mentioned. I am unsure if it is an error with the plugin or if it is an error with the linter. I am thinking it is an issue with the linter. I am still going to see what I can do but you can follow that in the following issue #46
